externals/kyuubi-chat-engine/src/main/scala/org/apache/kyuubi/engine/chat/operation/ChatOperation.scala

```diff
@@ -45,8 +46,11 @@ abstract class ChatOperation(session: Session) extends AbstractOperation(session
         iter.fetchAbsolute(0)
     }
 
-    val taken = iter.take(rowSetSize)
-    val resultRowSet = RowSet.toTRowSet(taken.toSeq, 1, getProtocolVersion)
+    val taken = iter.take(rowSetSize).map(_.toSeq)
+    val resultRowSet = new RowSetGenerator().toTRowSet(
```

Comment:
why should create a new instance each time? is it stateful? if not, I suppose we can define `object RowSetGenerator` instead of `class RowSetGenerator`

Reply:
It's designed on purpose. As for Spark's RowSetGenerator, we will have TimeFormatters instance inside, and should not be shared by others as they may not be thread-safe. As for Flink's RowSetGenerator, we init the RowSetGenerator with the timezone id instance.

Reply:
that makes sense. thanks for explanation